### PR TITLE
Model wrappers for functions

### PR DESCRIPTION
**Pull request by gbaydin**
Addresses #397

### CODE REVIEW

File: src/DiffSharp.Core/Model.fs [hunk 1]
```diff
@@ -491,12 +491,21 @@ type ModelBase() =
 
 
 /// <summary>Represents a model, primarily a collection of named parameters and sub-models and a function governed by them.</summary>
-[<AbstractClass>]
-type Model<'In, 'Out>() =
+// [<AbstractClass>]
+type Model<'In, 'Out>(?f:'In->'Out, ?parameters: seq<Parameter>, ?buffers: seq<Parameter>, ?models: seq<obj>) =
```
Comment (dsyme) [inline src/DiffSharp.Core/Model.fs]: Should that `obj` be `ModelBase`?

**Comment by dsyme** [comment]
Seems ok

File: src/DiffSharp.Core/Model.fs (hunk 1 above)
Comment (gbaydin) [reply src/DiffSharp.Core/Model.fs]: This design was to improve ease of use for `Model.addModel` etc. but I think it does make sense to get rid of this `obj` and use the correct type.

image

Comment (gbaydin) [reply src/DiffSharp.Core/Model.fs]: Fixed in a subsequent commit

**Comment by gbaydin** [comment]
I think this is ready to go.

**Pull request merged**